packages/picasso.js/src/core/chart-components/ref-line/lines-and-labels.js

```diff
@@ -225,8 +323,81 @@ export function createLineWithLabel({ chart, blueprint, renderer, p, settings, i
   // Always push the line,
   // but this is done after collision detection,
   // because otherwise it would collide with it's own line
-  items.push(line);
+  if (line) {
+    items.push(line);
+  } else if (slopeLine) {
+    // push slope line
+    items.push(slope);
+    // create data area labels for slope line
+    let valueString;
+    let rectLabel;
+    const maxLabelWidth = 120;
+    if (slopeLine.slope.value !== 0 && (slopeLine.showLabel || slopeLine.showValue)) {
+      let slopeLabelText = slopeLine.showLabel ? slopeLine.refLineLabel : '';
+      if (slopeLine.showValue) {
+        const formatter = getFormatter(p);
+        const formattedValue = formatter ? formatter(slopeLine.value) : slopeLine.value;
+        valueString = `(${slopeLine.slope.value}x + ${formattedValue})`;
+        slopeLabelText += slopeLine.refLineLabel ? ` ${valueString}` : valueString;
+      }
+      if (slopeLabelText.length > 1) {
+        // Measure the label text
+        let measured = renderer.measureText({
+          text: slopeLabelText,
+          fontFamily: slopeStyle.fontFamily,
+          fontSize: slopeStyle.fontSize,
+        });
+        measured.width = measured.width > maxLabelWidth ? maxLabelWidth : measured.width;
+        const xPadding =
+          slopeLine.slope.value > 0 && !slopeLine.isRtl ? slopeStyle.padding * 4 : slopeStyle.padding / 2;
+        const yPadding = slopeLine.slope.value > 0 ? slopeStyle.padding * 3 : slopeStyle.padding;
+        const maxX = isMaxY(chart, slopeLine.slope.value, slopeLine.value)
+          ? getMaxXPosition(chart, slopeLine.slope.value, slopeLine.value)
+          : undefined;
+        const maxY = maxX === undefined ? Math.abs(slope.y2) : 1;
 
+        const x = calculateX(slopeLine, slope, maxX, measured, blueprint, slopeStyle);
+        const y = slopeLine.slope.value > 0 ? maxY : slope.y1;
+        if (slopeLine.labelStroke) {
+          rectLabel = {
+            type: 'rect',
+            x: slopeLine.isRtl ? Math.max(x, xPadding) - 2 : Math.max(x, xPadding) + 2,
+            y: Math.abs(Math.max(y, yPadding) - measured.height),
+            rx: 3,
+            ry: 3,
+            width: measured.width + slopeStyle.padding,
+            height: measured.height - 2,
+            stroke: slopeStyle.background.stroke,
```

Comment:
Do we need stroke and strokeWidth here?

Reply:
It is defined in refLabelDefaultSettings in the file and is being used by the other rect created in the same file, so used the same as to opt for any stroke related styling ... but true not needed now... shall remove it